### ISSUE TITLE
Infiinite loop caused

### ISSUE THREAD

**Issue by developdeez** [i18next/react-i18next#673]
I keep getting infinite loop errors when using the HOC 'withNamespaces'
I put it on a child component of a page that is also translated.

**Comment by jamuhl** [comment]
please start adding changes related to the v9 version to the v9.x.x branch (master will be now for the upcoming v10 hooks)

--> https://github.com/i18next/react-i18next/tree/v9.x.x

**Comment by cthurston** [comment]
I'm getting an infinite loop when the backend fails to load.  Probably related, but I'm using alpha v10.

**Comment by jamuhl** [comment]
@cthurston with current published react-i18next@10.0.0-alpha.1? or an old v9

think i fixed that...https://github.com/i18next/react-i18next/commit/376fe57c02e2567a933039d3857df445bb3ba5bf#diff-cfd6b2fa22409acc10b6dab14d52cdb1R51

**Comment by cthurston** [comment]
I'm using the current published react-i18next@10.0.0-alpha.1.  For some reason it is sending and empty languages array.  This causes ready to always be false.  Near line 45 in useTranslation.  If I put a guard in the every callback, it seems not to crash.

````
var ready = i18n.isInitialized && namespaces.every(function (n) {
    if(!i18n.languages || !i18n.languages.length) return true
````

**Comment by jamuhl** [comment]
can you paste your i18next options? wondering how i18n.languages can be empty after init

**Comment by cthurston** [comment]
````
{
  fallbackLng: 'en',
  ns: ['default'],
  defaultNS: 'default',
  fallbackNS: 'default',
  nsSeparator: '::',
  keySeparator: '::',
  debug: true,
  interpolation: {
    escapeValue: false, // not needed for react!!
  },
  backend: {
    loadPath: `${REACT_APP_API_SERVER}/locales/{{lng}}/{{ns}}/translation.json`,
    crossDomain: true,
  },
}
````

**Comment by jamuhl** [comment]
strange if `isInitialized` is set on i18next languages should be at least `['en']` - really don't get how i18next.languages can be undefined

**Comment by jamuhl** [comment]
```js
  const ready =
    i18n.isInitialized &&
    namespaces.every(n => {
      const ret =
        i18n.hasResourceBundle(i18n.languages[0], n) ||
```

So guess it's an empty array...i'm confused

**Comment by cthurston** [comment]
````
var ready = i18n.isInitialized && namespaces.every(function (n) {
    console.log('Ready?', n, i18n.languages)
````

Without language detector:
<img width="730" alt="screen shot 2019-01-30 at 12 06 33 pm" src="https://user-images.githubusercontent.com/622081/52005752-afcad000-2487-11e9-9f92-d57e19186fcc.png">


But even with language detector and languages ['en', 'en-US'] it still hits an infinite loop if the backend doesn't respond.

**Comment by jamuhl** [comment]
@cthurston might be fixed in react-i18next@10.0.0-alpha.2

**Comment by jamuhl** [comment]
closing for now...feel free to provide a sample to reproduce.
